secboot/secboot_sb.go

```diff
@@ -167,3 +177,98 @@ func UnlockEncryptedVolumeWithRecoveryKey(name, device string) error {
 
 	return nil
 }
+
+func setupDeviceMapperTargetForDeviceUnlock(disk disks.Disk, name string) (string, string, error) {
+	// 1. mount name under the right mapper with 1mb offset
+	partUUID, err := disk.FindMatchingPartitionUUIDWithPartLabel(name)
```

Comment:
I think this is a problematic part of the PR. We're assuming a certain partition name, but this assumption is not documented in the code. AFAIU how our LUKS handling works, the partition (i.e. structure in gadget.yaml) can have any name, but we care about the filesystem labels inside it. Specifically, we create a LUKS volume, with a well known name `<fs-label>-enc`, which contains a filesystem with label `<fs-label>`. Then when unlocking we look for a partition that carries filesystem (in this case a LUKS volume) with name `<fs-label>-enc`, which upon unlocking makes a device with filesystem `<fs-label>` available in the system. In each of the cases, fs-label is well known for each structure with a given role, because we enforce it as part of validation (there's `checkImplicitLabel`).

In here, we have the encryption code during install, which creates a device that doesn't really carry any known characteristics, and then adds an intermediate layer of encryption. Now, in the unlock path we don't have the fs label, as the encryption is transparent, and there's no separate container for it (like a LUKS volume). So we must look for a specific partition name? Also, I'm not sure that if unlocked the block device would be properly scanned for a filesystem/partition table.